monkey/infection_monkey/monkey.py

```diff
@@ -175,8 +175,11 @@ def start(self):
                 if monkey_tunnel:
                     monkey_tunnel.set_tunnel_for_host(machine)
                 if self._default_server:
-                    machine.set_default_server(get_interface_to_target(machine.ip_addr) +
-                                               (':'+self._default_server_port if self._default_server_port else ''))
+                    if self._network.on_island(self._default_server):
```

Comment:
Why does this change things?

Reply:
If monkey is on the island, we want to pass the correct island's IP to the exploited host. If monkey is not on island, we don't want to pass current machine's IP as monkey island.